src/fabric/test/eunit/fabric_open_revs_test.erl

```diff
@@ -0,0 +1,213 @@
+        ],
+        open_revs_single(DbName, IdRevsOpts, [latest])
+    ).
+
+t_exact_rev_multiple_leafs({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{1, <<"z">>}]}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {1, [<<"z">>]}}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [])
+    ).
+
+t_exact_rev_latest_multiple_leafs({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{1, <<"z">>}]}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"x">>, <<"z">>]}}},
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"y">>, <<"z">>]}}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [latest])
+    ).
+
+t_all_revs_latest_single_leaf({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, all}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"x">>, <<"z">>]}}},
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"y">>, <<"z">>]}}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [latest])
+    ).
+
+t_two_revs_single_doc({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{2, <<"x">>}, {2, <<"y">>}]}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"x">>, <<"z">>]}}},
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"y">>, <<"z">>]}}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [])
+    ).
+
+t_two_revs_single_doc_one_missing({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{2, <<"y">>}, {1, <<"foo">>}]}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"y">>, <<"z">>]}}},
+            {{not_found, missing}, {1, <<"foo">>}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [])
+    ).
+
+t_two_revs_single_doc_latest({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{1, <<"z">>}, {2, <<"y">>}]}, []},
+    ?assertEqual(
+        [
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"x">>, <<"z">>]}}},
+            {ok, #doc{id = <<"a">>, revs = {2, [<<"y">>, <<"z">>]}}}
+        ],
+        open_revs_single(DbName, IdRevsOpts, [latest])
+    ).
+
+t_timeout_rexi_error({_, DbName}) ->
+    IdRevsOpts = {{<<"a">>, [{2, <<"x">>}]}, []},
+    meck:new(rexi_utils, [passthrough]),
+    meck:new(rexi, [passthrough]),
+    RecFun = fun(_Refs, _KeyPos, _Fun, St, _Timeout, _PerMsgTO) ->
+        {timeout, St}
+    end,
+    meck:expect(rexi_utils, recv, RecFun),
+    ?assertEqual({error, timeout}, fabric:open_revs(DbName, [IdRevsOpts], [])),
+    % Also check that workers were cleaned up properly
+    ?assertEqual(1, meck:num_calls(rexi, kill_all, 1)).
+
+open_revs_single(DbName, IdRevsOpts, Options) ->
+    {ok, [Res]} = fabric:open_revs(DbName, [IdRevsOpts], Options),
+    % Validate batched open_revs/3 call is equivalent to calling
+    % the single open_revs/4 with that same Id and Revs args.
+    {{Id, Revs}, DocOpts} = IdRevsOpts,
+    {ok, ResSingle} = fabric:open_revs(DbName, Id, Revs, DocOpts ++ Options),
+    ?assertEqual(ResSingle, Res),
```

Comment:
This is a very nice touch, although I'm guessing once we're convinced they are equivalent, the single version can be eliminated in favor of the more general function?

Reply:
At some point they could. But since we managed to keep both implementations around we're taking advantage of it testing various corner cases both for equivalence between the old and the new API.

Eventually the old version would just be re-implemented as `fabric:open_revs(Db, [{Id, Revs, DocOpts}], Options)` not unlike we already do for `update_doc` and `update_docs`